linter/lints/cabf_br/lint_crl_validity_period.go

```diff
@@ -42,17 +47,67 @@ func (l *crlValidityPeriod) CheckApplies(c *x509.RevocationList) bool {
 }
 
 func (l *crlValidityPeriod) Execute(c *x509.RevocationList) *lint.LintResult {
-	validity := c.NextUpdate.Sub(c.ThisUpdate)
-	if validity <= 0 {
+	/*
+	   Let's Encrypt issues two kinds of CRLs:
+
+	    1) CRLs containing subscriber certificates, created by crl-updater.
+	       These assert the distributionPoint and onlyContainsUserCerts
+	       boolean.
+	    2) CRLs containing issuer CRLs, created by the ceremony tool. These
+	       assert the onlyContainsCACerts boolean.
+
+	   We use the presence of these booleans to determine which BR-mandated
+	   lifetime to enforce.
+	*/
+
+	// The only way to determine which type of CRL we're dealing with. The
+	// issuingDistributionPoint must be parsed and the internal fields
+	// inspected.
+	idpOID := asn1.ObjectIdentifier{2, 5, 29, 28} // id-ce-issuingDistributionPoint
+	idpe := lints.GetExtWithOID(c.Extensions, idpOID)
+	if idpe == nil {
+		return &lint.LintResult{
+			Status:  lint.Warn,
+			Details: "CRL missing IDP",
+		}
+	}
+
+	// Step inside the outer issuingDistributionPoint sequence to get access to
+	// its constituent fields.
+	idpv := cryptobyte.String(idpe.Value)
+	_ = idpv.ReadASN1(&idpv, cryptobyte_asn1.SEQUENCE)
+
+	// Throw distributionPoint away.
+	_ = idpv.SkipOptionalASN1(cryptobyte_asn1.Tag(0).ContextSpecific().Constructed())
+
+	// Default to subscriber cert CRL.
+	var BRValidity = 10 * 24 * time.Hour
+	var validityString = "10 days"
+
+	// We read this boolean as a byte and ensure its value is 0xFF because
+	// cryptobyte.ReadASN1Boolean can't handle the custom encoding rules for the
+	// [1] and [2] tagged fields. For the purposes of this lint, we just need to
+	// know if the field exists.
+	var onlyContainsCACerts []byte
```

Comment:
If you follow the suggestion for additional refactoring I [mentioned here](https://github.com/letsencrypt/boulder/pull/7064#discussion_r1337781812), this comment can go away entirely, since the documentation will be on the helper function instead.

Reply:
That's because I forgot to use `ReadASN1BooleanWithTag` in that file 🤦🏼 . Thank you.

Reply:
That's because I forgot to use `lints.ReadASN1BooleanWithTag` in that file 🤦🏼. Good catch.